### ISSUE TITLE
ALTWFC has files for IRAJ - Japanese Pokemon BW

### ISSUE THREAD

**Issue by ScarlettHaze** [InternalLoss/DLS1-Files#2]
But there is no folder for IRAJ in the ALTWFC folder.

**Comment by Mythra** [comment]
This is expected, the Pokémon Black/White, and Pokémon Black 2/White 2 do not send their literal game id4, and instead send a single Game ID4, and use attributes in the list to filter down to a single gift. You can see the extra filtering in the `_list.txt` files.

So the events are there, just not where you might expect.

**Comment by G1fan** [comment]
> This is expected, the Pokémon Black/White, and Pokémon Black 2/White 2 do not send their literal game id4, and instead send a single Game ID4, and use attributes in the list to filter down to a single gift. You can see the extra filtering in the `_list.txt` files.
> 
> So the events are there, just not where you might expect.

Please correct me if I'm wrong. But as far as I can tell neither of the IRAO folders in the repo contain files listed alongside the appropriate attr1 identifier in the relevant _list for Japanese.

**Comment by Mythra** [comment]
We are in the process of regenerating missing events, and fixing some others. It is a well known bug there are gifts missing at this point in time.

**Comment by G1fan** [comment]
> We are in the process of regenerating missing events, and fixing some others. It is a well known bug there are gifts missing at this point in time.

I understand that there may be events missing, and thanks for all your work on fixing any issues. I just wanted to clarify for my own understanding how the gifts are retrieved.

My understanding is that several language versions (English, German, Spanish, etc) all advertise as IRAO and determine what gifts they receive based on the gift also having the appropriate attr1 tag for their language.
However Japanese and Korean specifically advertise as their own IDs (IRAJ, IRAK) and have their own lists of gifts that also all have the appropriate attr1 tag for that language in their relevant lists.

Am I correct in my understanding?

**Comment by Mythra** [comment]
This is not correct with how the AltWFC DLS1 server that is running in production works today (due to some patches applied on it, that were before my time). You are however correct that games underlying protocol traffic Game ID4 sending does work the way you describe (sometimes, some builds break this rule but it's generally true).

**Comment by G1fan** [comment]
> This is not correct with how the AltWFC DLS1 server that is running in production works today (due to some patches applied on it, that were before my time). You are however correct that games underlying protocol traffic Game ID4 sending does work the way you describe (sometimes, some builds break this rule but it's generally true).

Understood. Thanks for clearing that up for me. 👍
